yajsapi/runner/index.ts

```diff
@@ -719,21 +728,27 @@ export class Engine {
           for (let worker_task of [...workers]) {
             worker_task.cancel();
           }
-          // await asyncio.wait(workers, timeout=15, return_when=asyncio.ALL_COMPLETED)
+          await bluebird.Promise.any([
+            bluebird.Promise.all([...workers]),
+            promise_timeout(10)
+          ]);
         }
       } catch (error) {
         logger.error(error);
       }
-
-      // find_offers_task.cancel();
-    }
-
-    payment_closing = true;
-    if (agreements_to_pay) {
       await bluebird.Promise.any([
-        Promise.all([process_invoices_job]),
-        promise_timeout(15),
+        bluebird.Promise.all([
+          find_offers_task,
+          process_invoices_job,
+        ]),
+        promise_timeout(10)
       ]);
+      if (agreements_to_pay.size == 0) {
```

Comment:
In `yapapi` we wait for `process_invoices_job` only if `agreements_to_pay` is **not empty**.

Reply:
Good catch!

Reply:
Changed to > 0